### DISCUSSION TITLE
Does the use of tailwind-merge make the component client-only with nextjs?

### DISCUSSION THREAD

**Discussion by findawayer** [dcastil/tailwind-merge#485] [Help]
Invoking `twMerge` or a customized version with `extendTailwindMerge` both makes the React component client-only, when used within a Next.js application with SSR or SSG setup, apparently.

Nextjs server components needs to either:
- be marked as client-only with `'use client'` directive
- not use `twMerge`

I wonder what part of the library could possibly trigger such restrictions, or am I misusing the API? Does someone have similar experiance?

**Comment by dcastil** [comment]
Hey @findawayer! 👋

There is nothing in tailwind-merge that should force your components into client-only mode, it is just a bunch of JS functions which aren't using any environment-specific APIs. I'm not sure what's causing it, but maybe the file where you configured tailwind-merge somehow landed in a client-only import path? Meaning that the file is importing another file that hase`"use client"` in it?
